plugins/inputs/monit/monit_test.go

```diff
@@ -0,0 +1,495 @@
+// Copyright 2020, Verizon
+// Licensed under the terms of the MIT License. See LICENSE file in project root for terms.
+
+package monit
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMonit0(t *testing.T) {
```

Comment:
Let's set these service type tests up as a table test, and update the metric assertions to our latest standard.  Here is an example to help you get started:
```go
func TestServiceType(t *testing.T) {
	tests := []struct {
		name     string
		filename string
		expected []telegraf.Metric
	}{
		{
			name:     "check filesystem service type",
			filename: "testdata/status_response_0.xml",
			expected: []telegraf.Metric{
				testutil.MustMetric(
					"monit_filesystem",
					map[string]string{
					},
					map[string]interface{}{
						"status_code":            0,
						"monitoring_status_code": 0,
						"mode":                   0,
						"block_percent":          0.0,
						"block_usage":            0.0,
						"block_total":            0.0,
						"inode_percent":          0.0,
						"inode_usage":            0.0,
						"inode_total":            0.0,
					},
					time.Unix(0, 0),
				),
			},
		},
		{
			name:     "check directory service type",
			filename: "testdata/status_response_1.xml",
			expected: []telegraf.Metric{
				testutil.MustMetric(
					"monit_directory",
					map[string]string{
					},
					map[string]interface{}{
					},
					time.Unix(0, 0),
				),
			},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
				switch r.URL.Path {
				case "/_status":
					http.ServeFile(w, r, tt.filename)
				default:
					w.WriteHeader(http.StatusNotFound)
				}
			}))
			defer ts.Close()

			plugin := &Monit{
				Address: ts.URL,
			}

			plugin.Init()

			var acc testutil.Accumulator
			err := plugin.Gather(&acc)
			require.NoError(t, err)

			testutil.RequireMetricsEqual(t, tt.expected, acc.GetTelegrafMetrics(),
				testutil.IgnoreTime())
		})
	}
}

```

https://github.com/golang/go/wiki/TableDrivenTests

Reply:
Done. Please check.